pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -744,3 +747,111 @@ func TestOnDNSResponse(t *testing.T) {
 		})
 	}
 }
+func TestFQDNCacheMinTTL(t *testing.T) {
+	currentTime := time.Now()
+	testIP := "192.168.1.1"
+	testFQDN := "fqdn-test-pod.lfx.test"
+	tests := []struct {
+		name            string
+		expectedTTL     time.Time
+		fqdnCacheMinTTL uint32
+		dnsMsg          *dns.Msg
+	}{
+		{
+			name:            "Response TTL less than FQDNCacheTTL",
+			expectedTTL:     currentTime.Add(10 * time.Second),
+			fqdnCacheMinTTL: 10,
+			dnsMsg: &dns.Msg{
```

Comment:
There is a lot of repetition: we use the same DNS message every time as far as I can tell (with the only change being the ttl value).
One option is to use a closure such as this one:
```go
getDNSMsg := func(ttl in) *dns.Msg {
        return &dns.Msg{ ... }
}
```